library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -1360,8 +1361,8 @@ impl<T> VecDeque<T> {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn back_mut(&mut self) -> Option<&mut T> {
-        let len = self.len();
-        if !self.is_empty() { Some(&mut self[len - 1]) } else { None }
+        let last = self.len() - 1;
```

Comment:
Perhaps:
`self.get_mut(last.wrapping_sub(1))`

Reply:
Ah that's fair